text/0048-rename-max-mdb-size-var.md

```diff
@@ -0,0 +1,43 @@
+- Title: Rename MAX_MDB_SIZE  configuration variable
+- Start Date: 2021-
+- Specification PR: [#48](https://github.com/meilisearch/specifications/pull/48)
+- MeiliSearch Tracking-issues:
+
+
+# Rename MAX_MDB_SIZE configuration variable
+
+## 1. Functional Specification
+
+### I. Summary
+
+This specification is written to rename the `MAX_MDB_SIZE` configuration variable.
+
+### II. Motivation
+
+Since version 0.21 no longer has a main LMDB database for all indexes but one LMBD database per index, the concept of a `main` database no longer exists internally.
+
+
+### III. Additional Materials
+N/A
+
+### IV. Explanation
+
+ - Rename `--max-mdb-size` to `--max-index-size`
+ - Rename `MAX_MDB_SIZE` to `MAX_INDEX_SIZE`
+
+The size entered must therefore be greater than or equal to the largest index of the search engine.
+
+The known limitation is that smaller indexes will take up a space equal to the larger index.
```

Comment:
This is the theoretical `max` size, so smaller indexes won't get bigger.
The limitation is more about how these supposedly small indexes could get big if we push too many documents.

Reply:
Thanks @irevoire. I will update the specification!

Quick question, what happens if my space requirements exceed the size limit? Do I get an error message?

Reply:
Huum, I just tried to index a bunch of documents with a `max-mdb-size` of `0`, and it actually succeeds without error or warning :thinking: 
I'll have to let curqui or kero answer this one

Reply:
> Huum, I just tried to index a bunch of documents with a max-mdb-size of 0, and it actually succeeds without error or warning 🤔

Really weird! I think the `0` value is not handled. With v0.20.0
```bash
cargo run --release -- --max-mdb-size 40960
```
and adding a 10Mb dataset I get
```json
{
  "message": "Impossible to create index; heed error; MDB_MAP_FULL: Environment mapsize limit reached",
  "errorCode": "index_creation_failed",
  "errorType": "internal_error",
  "errorLink": "https://docs.meilisearch.com/errors#index_creation_failed"
}
```

If there is no space on the device when trying to launch MeiliSearch, the core engine is not launched and returns `No more space on the device`.

Reply:
I've tested on 0.21 at indexing time and I've got a successful HTTP code but with an infinite loop error on engine stdout side made of this error.

`
[2021-06-14T16:58:45Z DEBUG meilisearch_http::index_controller::index_actor::actor] Processing update 0
error while processing update: error with index: MDB_MAP_FULL: Environment mapsize limit reached`